### ISSUE TITLE
"ne" package backspace

### ISSUE THREAD

**Issue by italomaia** [termux/termux-packages#1085]
Backspace key is not set for the backage. Currently it does nothing.

**Comment by fornwall** [comment]
Does it actually work like the delete key (removing text to the right of the cursor)?

**Comment by italomaia** [comment]
Backspace key should work like in regular (non-vim) editors. Removes the character to the left.

**Comment by italomaia** [comment]
Could it be just an improper key mapping?

**Comment by craigbarnes** [comment]
This is just a bug in `ne`. Some terminals send `^H` (BS) for Backspace and some send `^?` (DEL). The only sane thing to do is map both of them to the erase left command by default, but `ne` only maps one of them. The correct mapping can't be deduced (from `$TERM`) like most others keys can, since it's configurable in many terminals and no two terminals that set `TERM=xterm*` can agree on which to use.

**Comment by craigbarnes** [comment]
It doesn't seem like the `ne` developers care to fix this. Someone reported the [issue](https://github.com/vigna/ne/issues/48) over a year ago and nothing has happened since. It seems like the better solution would be either to patch the package or use a different [editor](https://github.com/craigbarnes/dte/blob/master/config/binding/default#L51-L55).

**Comment by vigna** [comment]
As discussed elsewhere, not a bug in `ne`. Improper terminal emulators, and/or improper terminfo databases. Easily fixed by fixing the database. Sometimes the emulator has an option to sync its behavior with the terminfo database.

**Comment by craigbarnes** [comment]
> Improper terminal emulators, and/or improper terminfo databases

I agree with this, but it's reached a point where there's no longer an easy fix to the problem. Changing the terminfo database wouldn't fix the issue and would take years to get into distros anyway.

Out of all the possible solutions, the one with the fewest apparent downsides is for apps to bind both `^?` and `^H` to erase and not even attempt to determine which one is the "Backspace key". I've been through the full list of terminals that send `0x7F` for Delete (i.e. that would clash with this suggestion) and there's not a single one that isn't obsolete.

It's fair enough if you choose to continue as-is, but it's really not realistic at all to expect ncurses or the dozens of xterm clones to simply fix the issue at this stage.
